lib/View.js

```diff
@@ -112,34 +152,244 @@ class View extends Node {
 
+    if (!result.length) {
+      return NaN
+    }
+
     return parseInt(result[0].count.value)
   }
 
   static fromCube (cube) {
-    const cubeSource = CubeSource.fromSource(cube.source, cube)
+    if (!cube) {
+      throw Error('Needs a cube')
+    }
 
-    const view = new View({ parent: cube })
+    if (!cube.source) {
+      throw Error('Needs source')
+    }
 
+    cube.source.ptr
+      .addOut(ns.rdf.type, ns.view.CubeSource)
+      .addOut(ns.view.cube, toTerm(cube))
+
+    const view = new View({ parent: cube, source: cube.source })
     cube.dimensions.forEach(cubeDimension => {
-      view.addDimension(view.createDimension({
-        source: cubeSource,
-        path: cubeDimension.path,
-        as: cubeDimension.path
-      }))
+      const viewDimension = view.createDimension({
+        source: cube.source, path: cubeDimension.path, as: cubeDimension.path
+      })
+      view.addDimension(viewDimension)
     })
 
+    view.setDefaultColumns()
+
     return view
   }
+
+  setDefaultColumns () {
+    const orderingColumns = [...this.dimensions]
+    const sortAlphaNum = (a, b) => a.cubeDimensions[0].path.value.localeCompare(
+      b.cubeDimensions[0].path.value, 'en', { numeric: true })
+    orderingColumns.sort(sortAlphaNum)
+
+    const projectionPtr = this.ptr.out(ns.view.projection)
+    projectionPtr.addList(ns.view.columns,
+      orderingColumns.map(dimension => dimension.ptr.term))
+  }
+
+  orderBy (orderBy) {
+    const projectionPtr = this.ptr.out(ns.view.projection)
+
+    if (typeof orderBy === 'undefined') {
+      if (!projectionPtr.out(ns.view.orderBy).isList()) {
+        return null
+      }
+      const items = []
+      for (const current of projectionPtr.out(ns.view.orderBy).list()) {
+        items.push({ dimension: current.out(ns.view.dimension).term, direction: current.out(ns.view.direction).term })
+      }
+      return items
+    }
+
+    if (orderBy === null) {
+      if (projectionPtr.out(ns.view.orderBy).isList()) {
+        for (const current of projectionPtr.out(ns.view.orderBy).list()) {
+          current.deleteOut(ns.view.dimension)
+          current.deleteOut(ns.view.direction)
+        }
+        // This hangs clownface!
+        // projectionPtr.deleteList(ns.view.orderBy)
+        projectionPtr.deleteOut(ns.view.orderBy)
+      }
+    } else {
+      const items = []
+      for (const { dimension, direction } of orderBy) {
+        const item = rdf.blankNode()
+        this.dataset.add(rdf.quad(item, ns.view.dimension, dimension, this.graph))
+        this.dataset.add(rdf.quad(item, ns.view.direction, direction, this.graph))
+        items.push(item)
+      }
+      projectionPtr.addList(ns.view.orderBy, items)
+    }
+    return this
+  }
+
+  updateProjection ({ offset, limit, orderBy }) {
+    this.offset(null)
+    this.limit(null)
+    this.orderBy(null)
+    this.offset(offset)
+    this.limit(limit)
+    this.orderBy(orderBy)
+  }
+
+  get projectionDimensions () {
```

Comment:
Gets the dimensions ordered by the 'columns' criterion specified in the projection